pkg/querier/tripperware/query.proto

```diff
@@ -47,7 +47,10 @@ message Extent  {
 message SampleStream {
   repeated cortexpb.LabelPair labels = 1 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "metric", (gogoproto.customtype) = "github.com/cortexproject/cortex/pkg/cortexpb.LabelAdapter"];
   repeated cortexpb.Sample samples = 2 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "values"];
-  repeated SampleHistogramPair histograms = 3 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "histograms"];
+  repeated SampleHistogramPair histograms = 3 [(gogoproto.nullable) = true, (gogoproto.jsontag) = "histograms"];
+  // Full representation of native histograms and it should be always float histogram type.
+  // This field should be never deserialized into JSON and should only exist in protobuf format.
+  repeated cortexpb.Histogram rawHistograms = 4 [(gogoproto.nullable) = true, (gogoproto.jsontag) = "-"];
```

Comment:
I just wonder why the `rawHistograms` are needed in the `matrix` result.

Reply:
ok, thanks.

Reply:
I see. Could you merge it? Then, I start with the implementation for retrieving native histograms.